src/WebJobs.Script/HostOptionsProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class HostOptionsProvider : IHostOptionsProvider
+    {
+        private readonly JsonSerializer _serializer = JsonSerializer.Create(new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+        });
+
+        private readonly IEnumerable<IExtensionOptionsInfo> _extensionOptionsInfoCollection;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ILogger<HostOptionsProvider> _logger;
+        private readonly IOptionsMonitor<ConcurrencyOptions> _concurrencyOptions;
+
+        public HostOptionsProvider(IServiceProvider provider, IEnumerable<IExtensionOptionsInfo> extensionOptionsInfoCollection, IOptionsMonitor<ConcurrencyOptions> concurrencyOption, ILogger<HostOptionsProvider> logger)
```

Comment:
Looks like _serviceProvider is no longer used?

Reply:
Thanks. Removed and make sure test works.